### ISSUE TITLE
workflow manager (for WorkflowDbSource)

### ISSUE THREAD

**Issue by cornernote** [raoul2000/yii2-workflow#32]
Thought you may want to have a play with this:
https://github.com/cornernote/yii2-workflow-manager

It's based on your WorkflowDbSource gist:
https://gist.github.com/raoul2000/c38165b7e6b5f12a6945#file-workflowdbsource-php

I turned it into a module.  Would be great to see your feedback.

**Comment by cornernote** [comment]
Looks like this:

![image](https://cloud.githubusercontent.com/assets/51875/17660161/a351c124-6316-11e6-8e2b-28340fe6dc8d.png)

**Comment by raoul2000** [comment]
:astonished:
wow !! This seems very impressive work and be sure I will try it.

Having a quick look at the code, it brings me back to [your performance question](https://github.com/raoul2000/yii2-workflow/issues/27#issue-170625187) some days ago. Maybe one improvement would be to not use Yii2 ActiveRecord in the workflowDbSource component or at least return arrays instead AR instances. According to the [definitive Guide to Yii2](http://www.yiiframework.com/doc-2.0/guide-tutorial-performance-tuning.html#using-arrays) this could be an interesting optimization when dealing with a large amount of records...

**Comment by cornernote** [comment]
I implemented cache (copied from WorkflowFileSource), so the number of ActiveRecord/DB hits should be minimal for loading workflow data.  I am more concerned with instantiating all the workflow/status classes each time a model is loaded.

So far it's not been much overhead, however I need to try with larger jobs to get a better benchmark.

I submitted a PR to link to this from your readme, I hope that's ok.
